### PR TITLE
feat: Add Planck tapering functions for smoothing distributions

### PR DESCRIPTION
**Pull request by isaac-cf-wong**
## Summary by CodeRabbit

* **New Features**
  * Added a logarithmic variant of the Planck tapering function for stable log‑space computations.

* **Refactor**
  * Standardized type hints for tapering functions for consistent public signatures.

* **Tests**
  * Added comprehensive unit tests covering normal, boundary, and edge cases for both tapering functions.

### CODE REVIEW

File: src/gwsim_pop/config/main.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        Args:
+            filename: File name.
+            encoding: File encoding.
+
+        Returns:
+            Configuration.
+
+        """
+        data = read_yaml(filename=filename, encoding=encoding)
+        return cls(**data)
```
[the remark this answers is not in the document]
Comment (isaac-cf-wong) [reply src/gwsim_pop/config/main.py]: This will be fixed in a future PR.

File: tests/config/test_config_population.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        config = PopulationConfiguration(model="test_model")
+        assert config.model == "test_model"
+        assert isinstance(config.arguments, PopulationArguments)
+
+    def test_initialization_with_arguments(self):
+        """Test that PopulationConfiguration can be initialized with arguments."""
+        # Test with default arguments
+        config = PopulationConfiguration(model="test_model")
+        assert config.model == "test_model"
+        assert isinstance(config.arguments, PopulationArguments)
```
[the remark this answers is not in the document]
Comment (isaac-cf-wong) [reply tests/config/test_config_population.py]: This will be fixed in a future PR.

File: tests/config/test_config_selection.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        assert config.enabled is False
+        assert config.method == "snr"
+        assert config.arguments == {}
+
+    def test_initialization_with_default_values(self):
+        """Test that SelectionConfiguration can be initialized with default values."""
+        config = SelectionConfiguration()
+        assert config.enabled is False
+        assert config.method == "snr"
+        assert config.arguments == {}
```
[the remark this answers is not in the document]
Comment (isaac-cf-wong) [reply tests/config/test_config_selection.py]: This will be fixed in a future PR.
